docs/polkadot/implementors-guide/guide.md

```diff
@@ -0,0 +1,544 @@
+
+*on receiving CandidateBackingProcessMessage*
+* If the message is a `CandidateBackingProcessMessage::RegisterBackingWatcher`, register the watcher and trigger it each time a new candidate is backed. Also trigger it once initially if there are any backed candidates at the time of receipt.
+* If the message is a `CandidateBackingProcessMessage::Second`, sign and dispatch a `Seconded` statement only if we have not seconded any other candidate and have not signed a `Valid` statement for the requested candidate. Signing both a `Seconded` and `Valid` message is a double-voting misbehavior with a heavy penalty, and this could occur if another validator has seconded the same candidate and we've received their message before the internal seconding request.
+
+(TODO: send statements to Statement Distribution Process, handle shutdown signal from parent process)
+
+---
+
+### Candidate Proposal Process
+
+[TODO: get candidate from collator, feed to candidate backing process. ]
+
+### Secondary Checking
+
+[TODO]
+
+### Validator Assignment Process
+
+[TODO]
+
+### Collation Distribution Process
+
+[TODO]
+
+### Availability Distribution Process
+
+[TODO]
+
+----
+
+## Data Structures and Types
+
+[TODO]
+* CandidateReceipt
+* CandidateCommitments
+* AbridgedCandidateReceipt
+* GlobalValidationSchedule
+* LocalValidationData
+
+#### Block Import Event
+```rust
+/// Indicates that a new block has been added to the blockchain.
+struct BlockImportEvent {
+  /// The block header-hash.
+  hash: Hash,
+  /// The header itself.
+  header: Header,
+  /// Whether this block is considered the head of the best chain according to the 
+  /// event emitter's fork-choice rule.
+  new_best: bool,
+}
+```
+
+#### Block Finalization Event
+```rust
+/// Indicates that a new block has been finalized.
+struct BlockFinalizationEvent {
+  /// The block header-hash.
+  hash: Hash,
+  /// The header of the finalized block.
+  header: Header,
+}
+```
+
+#### Statement Type
+```rust
+/// A statement about the validity of a parachain candidate.
+enum Statement {
+  /// A statement about a new candidate.
+  Seconded(CandidateReceipt),
+  /// A statement about the validity of a candidate, based on candidate's hash.
+  Valid(Hash),
+  /// A statement about the invalidity of a candidate.
+  Invalid(Hash),
+}
+```
+
+#### Signed Statement Type
+
+This is a signed statement. The actual signed payload should reference only the hash of the CandidateReceipt and should include
+a relay parent which provides context to the signature. This prevents against replay attacks and allows the candidate receipt itself
+to be omitted when checking a signature on a `Seconded` statement.
+
+```rust
+/// A signed statement.
+struct SignedStatement {
+  statement: Statement,
+  signed: ValidatorId,
+  signature: Signature
+}
+```
+
+
+#### Overseer Signal
+
+Signals from the overseer to a process.
```

Comment:
`...to request change in execution that has to be obeyed by the process.` to clarify things a bit more based on the previous sections, maybe too verbose.